distributed/deploy/spec.py

```diff
@@ -413,7 +414,15 @@ async def _close(self):
             return
         if self.status == Status.running or self.status == Status.failed:
             self.status = Status.closing
-            self.scale(0)
+
+            # Need to call stop here before we close all servers to avoid having
+            # dangling tasks in the ioloop
+            with suppress(AttributeError):
+                self._adaptive.stop()
```

Comment:
This is a bit unfortunate since I need to call it up here and not down in `Cluster`. If I do this only in cluster, the even loop seems to close too soon and we have still pending tasks from `AdaptiveCore.adapt`.

I'm wondering if we ever considered adding `PYTHONASYNCIODEBUG=1` to our test suite which would raise in these instances. Not sure how much would break or if this is a bad idea in general